src/deepsparse/transformers/pipelines/text_generation/prep_for_generation.py

```diff
@@ -101,6 +101,7 @@ def run(
             else [],
             "finished_reason": [],
             "token_generator": token_generator,
+            "past_tokens_queue": copy.copy(tokens),
```

Comment:
is this just an initial copy of the prompt tokens, or does it happen for each iteration?

Reply:
Start with the prompt tokens. Update to add the new token and remove the first token in the queue. Basically a sliding window